Geometry/TrackerGeometryBuilder/src/TrackerGeomBuilderFromGeometricDet.cc

```diff
@@ -222,36 +224,66 @@ void TrackerGeomBuilderFromGeometricDet::buildSilicon(std::vector<const Geometri
 
 
 void TrackerGeomBuilderFromGeometricDet::buildGeomDet(TrackerGeometry* tracker){
+
   PlaneBuilderForGluedDet gluedplaneBuilder;
-  auto  const & gdu= tracker->detUnits();
-  auto  const & gduId = tracker->detUnitIds();
+  auto const & gdu = tracker->detUnits();
+  auto const & gduId = tracker->detUnitIds();
+  auto const & gduType = tracker->detTypes();
+
+  if(gdu.size() != gduId.size() || gdu.size() != gduType.size())
+    edm::LogError("BuildingGeomDetUnits") << "detUnits, detUnitIds and detTypes vectors in TrackerGeometry should be the same size!";
```

Comment:
@cvuosalo - thanks for spotting it. I think, it is a problem. It could be DB vs XML geometry mismatch.